src/ORM/Association/BelongsToMany.php

```diff
@@ -1329,9 +1329,18 @@ protected function _collectJointEntities(EntityInterface $sourceEntity, array $t
 
         $unions = [];
         foreach ($missing as $key) {
+            $conditions = array_merge(
+                array_combine($foreignKey, $sourceKey),
+                array_combine($assocForeignKey, $key)
+            );
+            foreach ($conditions as $k => $v) {
+                if ($v === null) {
+                    $conditions[$k . ' IS'] = $v;
+                    unset($conditions[$k]);
+                }
+            }
```

Comment:
I think I would prefer building 2 new arrays for the keys where they get `IS` appended, irregardless of the value they have. Then you can `array_combine` them with the values